### PR TITLE
Use tools image in Makefile

### PR DESCRIPTION
**Pull request by bkabrda**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,22 +1,21 @@
 .PHONY: dg doc build run test clean
 
+TOOLSIMG = modularitycontainers/image-build-tools
```
Comment (phracek) [inline Makefile]: I would prefer name it like `TOOLS_IMG`.

File: Makefile [hunk 2]
```diff
@@ -1,22 +1,21 @@
...
+TOOLSIMG_RM = true
```
Comment (phracek) [inline Makefile]: The same here

**Comment by phracek** [comment]
I will test it soon.

File: Makefile [hunk 3]
```diff
@@ -1,22 +1,21 @@
...
 DISTRO = fedora-26-x86_64
 VERSION = 2.4
 VARIANT = upstream
-DG = /usr/bin/dg
-GOMD2MAN = /usr/bin/go-md2man
 DOCKERFILE_SRC := Dockerfile.template
 DOCKERFILE := Dockerfile
 TEST_IMAGE_NAME := container-images-tests
 
+IMG_RUN_LINE = docker run --rm=${TOOLS_IMG_RM} -v /var/run/docker.sock:/var/run/docker.sock:Z -v ${PWD}:/src:Z ${TOOLS_IMG}
```
Comment (praiskup) [inline Makefile:12]: Seems like this approach expects that user is able to run containers, which means some non-default configuration of docker ; and a warning is that some people will refuse to configure it (docker community says that if you allow yourself to run containers => you are (almost) privileged as a root)

Comment (bkabrda) [reply Makefile:12]: Thanks for the comment. I added a commit that ammends README.md and adds some useful information about this approach. Hopefully that should be enough for now.

**Comment by bkabrda** [comment]
The CI seems to be failing only because problems in MTF. I believe that the PR itself is ok and can be merged now. @phracek can you please confirm?

**Comment by phracek** [comment]
Yes, bug is on MTF side. I am merging it.

**Pull request merged**